SingularityService/src/main/java/com/hubspot/singularity/auth/SingularityDualAuthorizer.java

```diff
@@ -0,0 +1,211 @@
+package com.hubspot.singularity.auth;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.hubspot.singularity.SingularityAuthorizationScope;
+import com.hubspot.singularity.SingularityRequest;
+import com.hubspot.singularity.SingularityUser;
+import com.hubspot.singularity.data.RequestManager;
+import javax.ws.rs.WebApplicationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Temporary class meant for comparing the newer groups + scopes auth to the older groups-only.
+ * This authorizer will use the response of the older groups auth, but log if the newer auth does
+ * not produce the same result.
+ * This class consumes the newer SingularityUser with both groups and scopes attached
+ */
+@Singleton
+public class SingularityDualAuthorizer extends SingularityAuthorizer {
+  private static final Logger LOG = LoggerFactory.getLogger(
+    SingularityDualAuthorizer.class
+  );
+
+  private final SingularityGroupsAuthorizer groupsAuthorizer;
+  private final SingularityGroupsScopesAuthorizer groupsScopesAuthorizer;
+
+  @Inject
+  public SingularityDualAuthorizer(
+    RequestManager requestManager,
+    boolean authEnabled,
+    SingularityGroupsAuthorizer groupsAuthorizer,
+    SingularityGroupsScopesAuthorizer groupsScopesAuthorizer
+  ) {
+    super(requestManager, authEnabled);
+    this.groupsAuthorizer = groupsAuthorizer;
+    this.groupsScopesAuthorizer = groupsScopesAuthorizer;
+  }
+
+  @Override
+  public boolean hasAdminAuthorization(SingularityUser user) {
+    boolean result = groupsAuthorizer.hasAdminAuthorization(user.withOnlyGroups());
+    if (result != groupsScopesAuthorizer.hasAdminAuthorization(user)) {
+      LOG.warn(
+        "Difference in auth of user {} for ADMIN, scopes authorizer: {}, groups authorizer: {}, user: {}",
+        user.getId(),
+        !result,
+        result,
+        user
+      );
+    }
+    return result;
+  }
+
+  @Override
+  public void checkAdminAuthorization(SingularityUser user) {
```

Comment:
Looks like this and several other methods below follow the same patten of checkGrantedByScopes -> check groups authorizer -> maybe warn. Is there a way to factor the pattern out?

Reply:
yeah, I can take another pass at slimming it down after I do a few initial tests in staging. Did one pass already to factor some things out, but I think you're right that I can DRY it up a bit more